docs/references/file_provider_markdown.md

```diff
@@ -0,0 +1,272 @@
+
+An empty configuration file contains only a `Configuration` tag, and a
+configuration file with user input has a tag corresponding to each value
+that a user sets. See the example below.
+
+```xml
+<Configuration>
+    <Value>10</Value>
+</Configuration>
+```
+
+### Workflow Configuration File
+
+Each kind of example tool has its own XML configuration file. The file
+contains tool information like name, description, input anchors, and
+output anchors. The input and output anchor information is found under
+the `InputConnections` and `OutputConnections` tags respectively. Each
+`Connection` corresponds to 1 anchor. If the `AllowMultiple` flag is
+`True` then there can be multiple connections attached to that anchor.
+
+### Metadata File
+
+The metadata XML file contains information about each field of the
+incoming data. Each field must have a `name`, `size`, and `type`.
+
+-   `name`: The field name.
+-   `size`: The size of each piece of data in that field.
+-   `type`: The type of each piece of data in that field.
+
+`type` must correspond to 1 of the values in the `FieldType` enum. See
+the example below:
+
+```xml
+<RecordInfo>
+  <Field name="Name" size="254" type="v_string" />
+  <Field name="Value" size="254" type="double" scale="2" description="the value" source="internet"/>
+</RecordInfo>
+```
+
+### Record File
+
+The record CSV file contains all of the record information. It looks
+like a CSV file that Designer might use as input. The first row should
+contain field names corresponding to the name of each field in the
+metadata file. The rest of the rows should contain the record data. See
+the example below:
+
+```cs
+Name,Value
+Ten,10
+Twenty,20
+Thirty,30
+```
+
+Running the File Provider
+-------------------------
+
+The file provider is run from a command-line argument. The only argument
+the user must pass in is a path to a JSON file with this format:
+
+1.  `tool`: This is a dictionary with 2 keys: `plugin` and `path`.
+
+     -   `plugin`: Contains a string that corresponds to the class
+         plugin name.
+     -   `path`: Contains a string that corresponds to the folder that
+         contains the plugin, in other words,
+         `{"plugin": "Example", "path": "path/to/ExampleFolder"}`.
+
+2.  `tool\_config`: This is a string that corresponds to the absolute
+    path of the tool\'s configuration file, in other words,
+    `path/to/ExampleToolConfig.xml`.
+3.  `workflow\_config`: This is a string that corresponds to the
+    absolute path of the workflow\'s configuration file, in other words,
+    `path/to/ExampleWorkflowConfig.xml`.
+4.  `inputs`: This is a list of dictionaries that contain information
+    about the tool\'s input connections. It is optional since not all
+    tools have input. The user should specify input information for each
+    input connection that is associated with this tool. Each input
+    dictionary has 3 keys: `anchor_name`, `records`, and `metadata`.
```

Comment:
TW review:

`inputs`: This is a list of dictionaries that contain information
    about the tool\'s input connections. It's optional since not all
    tools have input. You should specify input information for each
    input connection that is associated with this tool. Each input
    dictionary has 3 keys: `anchor_name`, `records`, and `metadata`.

Reply:
fixed